src/nodes/import.ts

```diff
@@ -150,3 +151,31 @@ const collectInlineTypeImports = elements => {
 
   return { typeImports, regularImports };
 };
+
+const getNamedBindings = withEnv<
+  { recordFactory: boolean },
+  [RawNode],
+  ts.NamedImportBindings
+>((env, node) => {
+  if (node.importClause) {
+    const bindings = node.importClause.namedBindings;
+    if (env.recordFactory && node.moduleSpecifier.text === "immutable") {
+      // When we have encountered the `Record.Factory` type,
+      // we rename it to `RecordFactory` to match the flow typedefs.
+      // Since `RecordFactory` doesn't exist in TypeScript,
+      // we need to add an import for it here.
+      // We add the TYPE_KEYWORD first so that it becomes an "inline type import"
```

Comment:
Would an alternative to this be to declare `Record.Factory` as an alias of `RecordFactory` in our Flow type defs for immutable?

Reply:
I tried this, and was unable to come up with a syntax that worked.

Reply:
Further details:

You can't `declare namespace` in a `.js.flow` file, and you also can't seem to declare a `static` class property that takes a generic type argument.